addons/website_sale/tests/test_website_sale_pricelist.py

```diff
@@ -416,6 +418,39 @@ def test_pricelist_item_validity_period(self):
             self.assertEqual(sol.price_unit, 80.0, 'Reduction should be applied')
             self.assertEqual(sol.price_total, 160)
 
+    def test_pricelist_anonymous_user(self):
+        list_benelux_2 = self.list_benelux.sudo().copy({
+            'name': 'Benelux 2',
+            'item_ids': [
+                Command.create({
+                    'compute_price': 'percentage',
+                    'base': 'list_price',
+                    'percent_price': 20,
+                }),
+            ]
+        })
+        public_user = self.env.ref('base.public_user')
+        order_sudo = self.env['sale.order'].sudo().create({
+            'partner_id': public_user.partner_id.id,
+            'pricelist_id': list_benelux_2.id,
+            'order_line': [(0, 0, {
+                'name': self.product.name,
+                'product_id': self.product.id,
+                'product_uom_qty': 1,
+                'price_unit': self.product.list_price,
+                'tax_id': False,
+            })],
+        })
+        # Creating partner with address of belgium
+        partner = self.env['res.partner'].create({
+            'name': 'Test Partner',
+            'company_id': False,
+            'country_id': self.env.ref('base.be').id,
+        })
+        with MockRequest(self.env, website=self.website, website_sale_current_pl=list_benelux_2.id, website_sale_selected_pl=list_benelux_2.id) as request:
```

Comment:
```suggestion
       website = website.with_user(self.public_user)
        with MockRequest(website.env, website=website, website_sale_current_pl=list_benelux_2.id, website_sale_selected_pl=list_benelux_2.id) as request:
```

Make sure to mimic an environment that would be (nearly) identical to the one used in the flow you're testing ;)

Reply:
Sure!